src/kOS/Suffixed/WaypointValue.cs

```diff
@@ -8,47 +9,80 @@ namespace kOS.Suffixed
 {
     public class WaypointValue : Structure
     {
-        protected Waypoint WayPoint { get; set; }
+        protected Waypoint WrappedWaypoint { get; set; }
         protected CelestialBody CachedBody { get; set; }
         protected SharedObjects Shared { get; set; }
+        private static Dictionary<string,int> greekMap;
         
         public WaypointValue(Waypoint wayPoint, SharedObjects shared)
         {
-            WayPoint = wayPoint;
+            WrappedWaypoint = wayPoint;
             Shared = shared;
             AddSuffix("DUMP", new NoArgsSuffix<string>(ToVerboseString)); // for debugging
-            AddSuffix("NAME", new NoArgsSuffix<string>(() => wayPoint.name, "Name of waypoint as it appears on the map and contract"));
-            AddSuffix("BODY", new NoArgsSuffix<BodyTarget>(() => new BodyTarget(GetBody(),shared), "Celestial body the waypoint is attached to"));
+            AddSuffix("NAME", new NoArgsSuffix<string>(() => CookedName(), "Name of waypoint as it appears on the map and contract"));
+            AddSuffix("BODY", new NoArgsSuffix<BodyTarget>(() => new BodyTarget(GetBody(),Shared), "Celestial body the waypoint is attached to"));
```

Comment:
Esoterically, I prefer using the class member here ```Shared``` (capitalized) as opposed to the constructor's argument ```shared``` (lowercase).  Granted, they're the same at this point, but It just seems more appropriate as it makes the code more mobile (it could be moved outside the constructor and still work, this way).